File: Scala/ProblemSolving/The Hurdle Race.scala
import java.io._
import scala.io._


object Result {

    def hurdleRace(k: Int, height: Array[Int]): Int = {
        val maxi = height.distinct.sorted.last
    if(maxi - k > 0){
      maxi - k
    }
    else{
      0
    }

    }

}

object Solution {
    def main(args: Array[String]) {
        val printWriter = new PrintWriter(sys.env("OUTPUT_PATH"))

        val firstMultipleInput = StdIn.readLine.replaceAll("\\s+$", "").split(" ")

        val n = firstMultipleInput(0).toInt

        val k = firstMultipleInput(1).toInt

        val height = StdIn.readLine.replaceAll("\\s+$", "").split(" ").map(_.trim.toInt)

        val result = Result.hurdleRace(k, height)

        printWriter.println(result)

        printWriter.close()
    }
}

File: Python/ProblemSolving/The Hurdle Race.py
#!/bin/python3

import math
import os
import random
import re
import sys

def hurdleRace(k, height):
    maxi = max(height)
    if maxi - k > 0:
        return maxi - k
    else:
        return 0

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    first_multiple_input = input().rstrip().split()

    n = int(first_multiple_input[0])

    k = int(first_multiple_input[1])

    height = list(map(int, input().rstrip().split()))

    result = hurdleRace(k, height)

    fptr.write(str(result) + '\n')

    fptr.close()
